pkg/clusterversion/cv.go

```diff
@@ -275,8 +275,8 @@ func (c *clusterVersionClient) runUpgradeWithImage(cv *configv1.ClusterVersion,
 
 	if cv.Spec.DesiredUpdate == nil || cv.Spec.DesiredUpdate.Image != desired.Image {
 		logger.Info(fmt.Sprintf("Setting ClusterVersion to Image %s", desired.Image))
-		cv.Spec.DesiredUpdate = &configv1.Update{Image: desired.Image, Version: ""}
-		err := c.client.Update(context.TODO(), cv)
+		patch := []byte(fmt.Sprintf(`{"spec":{"desiredUpdate":{"image":"%s","version":null}}}`, desired.Image))
```

Comment:
not a big deal, but maybe better to change the name here to something like `desiredImage` instead of `patch`?

Reply:
same for below, like desiredChannel, desiredVersion

Reply:
That's a good idea.
Thanks for the suggestion.